app/src/swig/app.i

```diff
@@ -832,6 +878,27 @@ static firebase::AppOptions* AppOptionsLoadFromJsonConfig(const char* config) {
     Firebase.Platform.FirebaseHandler.Create(
         Firebase.Platform.FirebaseAppUtils.Instance);
     FirebaseApp newProxy;
+
+    // If this is the first App, register library information.
+    if (!userAgentRegistered) {
+        userAgentRegistered = true;
+        var userAgentMap = new StringStringMap();
```

Comment:
since 'new' is used here, I think the map should be deleted after the call to RegisterLibrariesInternal

Reply:
This is C# code block so userAgentMap would be simply GCed.